src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/Filtering/DocIdSymbolFilter.cs

```diff
@@ -9,9 +9,15 @@ namespace Microsoft.DotNet.ApiSymbolExtensions.Filtering
     /// Implements the logic of filtering out api.
     /// Reads the file with the list of attributes, types, members in DocId format.
     /// </summary>
-    public class DocIdSymbolFilter(string[] docIdsToExcludeFiles) : ISymbolFilter
+    public class DocIdSymbolFilter : ISymbolFilter
     {
-        private readonly HashSet<string> _docIdsToExclude = new(ReadDocIdsAttributes(docIdsToExcludeFiles));
+        private readonly HashSet<string> _docIdsToExclude;
+
+        public static DocIdSymbolFilter GetFilterForDocIds(params string[] docIdsToExclude)
```

Comment:
Could this just be the class's constructor?

Reply:
I guess it could, but I've seen that normally the preference is to have constructors be used for initialization only, and instance methods are used to do actual functionality. Plus that's what we were doing before, and the method name describes what it's doing (the constructor name does not).

Reply:
We prefer to use a static method when we do more work that what is expected in a constructor. Looking at the code here it looks like we are opening and reading files which definitely shouldn't happen in a constructor. So I agree, the static method makes more sense here.

IMO `Create` as the static method name makes more sense. Example: https://github.com/dotnet/sdk/blob/7361d854bbdb8681556ea579a5c1e2c4e1ffb051/src/Compatibility/ApiCompat/Microsoft.DotNet.PackageValidation/Package.cs#L123

In that case you invoke `Package.Create`. Here you would invoke `DoIdSymbolFilter.Create` which I think is also descriptive enough.

Reply:
Alright, I can change it to Create.

Reply:
Done.